compile issue on SupermucNG

**Describe the bug**
Error during compilation:

```
[ 41%] Building CXX object CMakeFiles/SeisSol-common-lib.dir/src/ResultWriter/AnalysisWriter.cpp.o
/dss/lrzsys/sys/spack/release/22.2.1/opt/x86_64/gcc/11.2.0-gcc-bmnv5vk/include/c++/11.2.0/bits/unique_ptr.h(83): error: incomplete type is not allowed
  	static_assert(sizeof(_Tp)>0,
  	                     ^
          detected during:
            instantiation of "void std::default_delete<_Tp>::operator()(_Tp *) const [with _Tp=seissol::kernels::PointSourceCluster]" at line 361
            instantiation of "std::unique_ptr<_Tp, _Dp>::~unique_ptr() noexcept [with _Tp=seissol::kernels::PointSourceCluster, _Dp=std::default_delete<seissol::kernels::PointSourceCluster>]" at line 59 of "/dss/dsshome1/07/ru76tuj2/SeisSol_20240607/src/SourceTerm/typedefs.hpp"

[ 42%] Building CXX object CMakeFiles/SeisSol-common-lib.dir/src/SourceTerm/NRFReader.cpp.o
compilation aborted for /dss/dsshome1/07/ru76tuj2/SeisSol_20240607/src/SourceTerm/PointSource.cpp (code 2)
make[2]: *** [CMakeFiles/SeisSol-common-lib.dir/build.make:706: CMakeFiles/SeisSol-common-lib.dir/src/SourceTerm/PointSource.cpp.o] Error 2
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:290: CMakeFiles/SeisSol-common-lib.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

**Expected behavior**
Code should compile.

**To Reproduce**
Steps to reproduce the behavior:
1. Which version do you use? Provide branch and commit id.
main 60aedc8c2ae0a5d9e5a46cc3a40e4eaf1c047225
2. Which build settings do you use? Which compiler version do you use?
`cmake -DNUMA_AWARE_PINNING=ON -DASAGI=ON -DCMAKE_BUILD_TYPE=Release -DHOST_ARCH=skx -DPRECISION=double -DORDER=4 -DGEMM_TOOLS_LIST=PSpaMM ..`
4. On which machine does your problem occur? If on a cluster: Which modules are loaded?
Documented supermucNG spack build:
![image](https://github.com/SeisSol/SeisSol/assets/42612980/8bcbe783-6188-48cf-9019-4d9e11ed7f4d)


**Additional context**
Compilation works with commit 6d301757378ad8446173e0a12c095a695a708aaf

--- a/src/Kernels/PointSourceCluster.h
+++ b/src/Kernels/PointSourceCluster.h
@@ -13,6 +13,7 @@
 #include <algorithm>
 #include <cmath>
 #include <cstdint>
+#include <memory>
 
 namespace seissol::kernels {
 class PointSourceCluster {
@@ -21,6 +22,11 @@ class PointSourceCluster {
   virtual void addTimeIntegratedPointSources(
       double from, double to, seissol::parallel::runtime::StreamRuntime& runtime) = 0;
   virtual unsigned size() const = 0;
+};
+
+struct PointSourceClusterPair {
+  std::unique_ptr<kernels::PointSourceCluster> host{nullptr};
+  std::unique_ptr<kernels::PointSourceCluster> device{nullptr};
 };
 
 /**

--- a/src/Solver/time_stepping/TimeCluster.cpp
+++ b/src/Solver/time_stepping/TimeCluster.cpp
@@ -132,7 +132,7 @@ seissol::time_stepping::TimeCluster::TimeCluster(unsigned int i_clusterId, unsig
     frictionSolver(i_FrictionSolver),
     frictionSolverDevice(i_FrictionSolverDevice),
     faultOutputManager(i_faultOutputManager),
-    m_sourceCluster(sourceterm::PointSourceClusterPair{nullptr, nullptr}),
+    m_sourceCluster(seissol::kernels::PointSourceClusterPair{nullptr, nullptr}),
     // cells
     m_loopStatistics(i_loopStatistics),
     actorStateStatistics(actorStateStatistics),
@@ -176,7 +176,7 @@ seissol::time_stepping::TimeCluster::~TimeCluster() {
 }
 
 void seissol::time_stepping::TimeCluster::setPointSources(
-    seissol::sourceterm::PointSourceClusterPair sourceCluster) {
+    seissol::kernels::PointSourceClusterPair sourceCluster) {
   m_sourceCluster = std::move(sourceCluster);
 }
 

--- a/src/Solver/time_stepping/TimeCluster.h
+++ b/src/Solver/time_stepping/TimeCluster.h
@@ -173,7 +173,7 @@ private:
     dr::friction_law::FrictionSolver* frictionSolverDevice;
     dr::output::OutputManager* faultOutputManager;
 
-    seissol::sourceterm::PointSourceClusterPair m_sourceCluster;
+    seissol::kernels::PointSourceClusterPair m_sourceCluster;
 
     enum class ComputePart {
       Local = 0,
@@ -348,7 +348,7 @@ public:
    *
    * @param sourceCluster Contains point sources for cluster
    */
-  void setPointSources(seissol::sourceterm::PointSourceClusterPair sourceCluster);
+  void setPointSources(seissol::kernels::PointSourceClusterPair sourceCluster);
   void freePointSources() { m_sourceCluster.host.reset(nullptr); m_sourceCluster.device.reset(nullptr); }
 
   void setReceiverCluster( kernels::ReceiverCluster* receiverCluster) {

--- a/src/Solver/time_stepping/TimeManager.cpp
+++ b/src/Solver/time_stepping/TimeManager.cpp
@@ -354,7 +354,7 @@ double seissol::time_stepping::TimeManager::getTimeTolerance() {
 }
 
 void seissol::time_stepping::TimeManager::setPointSourcesForClusters(
-    std::unordered_map<LayerType, std::vector<seissol::sourceterm::PointSourceClusterPair>> sourceClusters) {
+    std::unordered_map<LayerType, std::vector<seissol::kernels::PointSourceClusterPair>> sourceClusters) {
   for (auto& cluster : clusters) {
     auto layerClusters = sourceClusters.find(cluster->getLayerType());
     if (layerClusters != sourceClusters.end() && cluster->getClusterId() < layerClusters->second.size()) {

--- a/src/Solver/time_stepping/TimeManager.h
+++ b/src/Solver/time_stepping/TimeManager.h
@@ -165,7 +165,7 @@ class seissol::time_stepping::TimeManager {
      * @param sourceClusters Collection of point sources for clusters
      */
     void setPointSourcesForClusters(
-        std::unordered_map<LayerType, std::vector<seissol::sourceterm::PointSourceClusterPair>> sourceClusters);
+        std::unordered_map<LayerType, std::vector<seissol::kernels::PointSourceClusterPair>> sourceClusters);
 
   /**
    * Returns the writer for the receivers

--- a/src/SourceTerm/Manager.cpp
+++ b/src/SourceTerm/Manager.cpp
@@ -288,7 +288,8 @@ auto makePointSourceCluster(ClusterMapping mapping,
                             const unsigned* meshIds,
                             seissol::initializer::LTSTree* ltsTree,
                             seissol::initializer::LTS* lts,
-                            seissol::initializer::Lut* ltsLut) -> PointSourceClusterPair {
+                            seissol::initializer::Lut* ltsLut)
+    -> seissol::kernels::PointSourceClusterPair {
   auto hostData = std::pair<std::shared_ptr<ClusterMapping>, std::shared_ptr<PointSources>>(
       std::make_shared<ClusterMapping>(mapping), std::make_shared<PointSources>(sources));
 
@@ -319,7 +320,7 @@ auto makePointSourceCluster(ClusterMapping mapping,
   auto deviceData = hostData;
 #endif
 
-  return PointSourceClusterPair{
+  return seissol::kernels::PointSourceClusterPair{
       std::make_unique<kernels::PointSourceClusterOnHost>(hostData.first, hostData.second),
       std::make_unique<GpuImpl>(deviceData.first, deviceData.second)};
 }
@@ -330,7 +331,7 @@ auto loadSourcesFromFSRM(const char* fileName,
                          seissol::initializer::LTS* lts,
                          seissol::initializer::Lut* ltsLut,
                          seissol::memory::Memkind memkind)
-    -> std::unordered_map<LayerType, std::vector<PointSourceClusterPair>> {
+    -> std::unordered_map<LayerType, std::vector<seissol::kernels::PointSourceClusterPair>> {
   // until further rewrite, we'll leave most of the raw pointers/arrays in here.
 
   int rank = seissol::MPI::mpi.rank();
@@ -362,7 +363,8 @@ auto loadSourcesFromFSRM(const char* fileName,
   logInfo(rank) << "Mapping point sources to LTS cells...";
   auto layeredClusterMapping =
       mapPointSourcesToClusters(meshIds.data(), numSources, ltsTree, lts, ltsLut, memkind);
-  std::unordered_map<LayerType, std::vector<PointSourceClusterPair>> layeredSourceClusters;
+  std::unordered_map<LayerType, std::vector<seissol::kernels::PointSourceClusterPair>>
+      layeredSourceClusters;
 
   for (auto layer : {Interior, Copy}) {
     auto& sourceCluster = layeredSourceClusters[layer];
@@ -440,7 +442,7 @@ auto loadSourcesFromNRF(const char* fileName,
                         seissol::initializer::LTS* lts,
                         seissol::initializer::Lut* ltsLut,
                         seissol::memory::Memkind memkind)
-    -> std::unordered_map<LayerType, std::vector<PointSourceClusterPair>> {
+    -> std::unordered_map<LayerType, std::vector<seissol::kernels::PointSourceClusterPair>> {
   int rank = seissol::MPI::mpi.rank();
 
   logInfo(rank) << "Reading" << fileName;
@@ -483,7 +485,8 @@ auto loadSourcesFromNRF(const char* fileName,
 
   auto layeredClusterMapping =
       mapPointSourcesToClusters(meshIds.data(), numSources, ltsTree, lts, ltsLut, memkind);
-  std::unordered_map<LayerType, std::vector<PointSourceClusterPair>> layeredSourceClusters;
+  std::unordered_map<LayerType, std::vector<seissol::kernels::PointSourceClusterPair>>
+      layeredSourceClusters;
 
   for (auto layer : {Interior, Copy}) {
     auto& sourceCluster = layeredSourceClusters[layer];
@@ -554,7 +557,8 @@ void Manager::loadSources(seissol::initializer::parameters::PointSourceType sour
 #else
   auto memkind = seissol::memory::Standard;
 #endif
-  auto sourceClusters = std::unordered_map<LayerType, std::vector<PointSourceClusterPair>>{};
+  auto sourceClusters =
+      std::unordered_map<LayerType, std::vector<seissol::kernels::PointSourceClusterPair>>{};
   if (sourceType == seissol::initializer::parameters::PointSourceType::NrfSource) {
     logInfo(seissol::MPI::mpi.rank()) << "Reading an NRF source (type 42).";
 #if defined(USE_NETCDF) && !defined(NETCDF_PASSIVE)

--- a/src/SourceTerm/typedefs.hpp
+++ b/src/SourceTerm/typedefs.hpp
@@ -48,18 +48,8 @@
 #include "generated_code/tensor.h"
 #include <array>
 #include <cstdlib>
-#include <memory>
-
-namespace seissol::kernels {
-class PointSourceCluster;
-} // namespace seissol::kernels
 
 namespace seissol::sourceterm {
-struct PointSourceClusterPair {
-  std::unique_ptr<kernels::PointSourceCluster> host{nullptr};
-  std::unique_ptr<kernels::PointSourceCluster> device{nullptr};
-};
-
 /** Models point sources of the form
  *    S(xi, eta, zeta, t) := (1 / |J|) * S(t) * M * delta(xi-xi_s, eta-eta_s, zeta-zeta_s),
  * where S(t) : t -> \mathbb R is the moment time history,

